### ISSUE TITLE
Metrics on derived quantities not available in training/validation steps

### ISSUE THREAD

**Issue by mcgibbon** [ai2cm/ace#224]
Computing derived quantities currently only happens during the inference loop, meaning we don't get to see e.g. the validation bias of surface pressure due to dry air or the single-sample error of the moisture budget residual.

This can be resolved by computing derived quantities after computing losses but before passing the data to aggregators, in the stepper loop.

**Comment by oliverwm1** [comment]
Agreed there would be value to having validation statistics about the derived quantities. The main downside IMO is increasing complexity of the stepper. `run_on_batch` is getting quite long already. I'm also conscious of the memory usage coming from having more and more derived quantities, but this is a somewhat separate issue from where they are computed.

**Comment by gideonite** [comment]
Any idea what effect this would have on training time?

**Comment by mcgibbon** [comment]
Another option would be to compute them after stepping in the train, validation, and inference code, but if we're always going to want these (aren't we?) it should probably go in the stepper. On the other hand, if we don't want to add it to training because of concerns about training time (I don't have numbers, I'd have to test it), I could add this to validation only.

**Comment by oliverwm1** [comment]
Unless I'm mistaken, we don't track any metrics computed from the training data except the loss. So it seems fine to only compute these things for the validation/inference case.

**Comment by mcgibbon** [comment]
Great, I'll add this to the validation step instead of the stepper, then.
